core/sail/inferencer/src/main/java/org/eclipse/rdf4j/sail/inferencer/fc/FastRdfsForwardChainingSailConnection.java

```diff
@@ -0,0 +1,1026 @@
+
+            @Override
+            public boolean remove(Resource resource, IRI iri, Value value, Resource... resources) {
+                return false;
+            }
+
+            @Override
+            public Model filter(Resource resource, IRI iri, Value value, Resource... resources) {
+                return null;
+            }
+
+            @Override
+            public Set<Resource> subjects() {
+                return null;
+            }
+
+            @Override
+            public Set<IRI> predicates() {
+                return null;
+            }
+
+            @Override
+            public Set<Value> objects() {
+                return null;
+            }
+
+            @Override
+            public ValueFactory getValueFactory() {
+                return null;
+            }
+
+            @Override
+            public Iterator<Statement> match(Resource resource, IRI iri, Value value, Resource... resources) {
+                return null;
+            }
+
+            @Override
+            public int size() {
+                return 0;
+            }
+
+            @Override
+            public boolean isEmpty() {
+                return false;
+            }
+
+            @Override
+            public boolean contains(Object o) {
+                return false;
+            }
+
+            @Override
+            public Iterator<Statement> iterator() {
+                return null;
+            }
+
+            @Override
+            public Object[] toArray() {
+                return new Object[0];
+            }
+
+            @Override
+            public <T> T[] toArray(T[] a) {
+                return null;
+            }
+
+            @Override
+            public boolean add(Statement statement) {
+                return false;
+            }
+
+            @Override
+            public boolean remove(Object o) {
+                return false;
+            }
+
+            @Override
+            public boolean containsAll(Collection<?> c) {
+                return false;
+            }
+
+            @Override
+            public boolean addAll(Collection<? extends Statement> c) {
+                return false;
+            }
+
+            @Override
+            public boolean removeAll(Collection<?> c) {
+                return false;
+            }
+
+            @Override
+            public boolean retainAll(Collection<?> c) {
+                return false;
+            }
+
+            @Override
+            public void clear() {
+
+            }
+        };
+    }
+
+
+    protected int applyRules(Model model) throws SailException {
+
+
+        return 0;
+    }
+
+
+    public void addStatement(Resource subject, IRI predicate, Value object, Resource... contexts) throws SailException {
+        addStatement(true, subject, predicate, object, contexts);
+    }
+
+
+    public void addStatement(boolean actuallyAdd, Resource subject, IRI predicate, Value object, Resource... resources) throws SailException {
```

Comment:
Should be private. It's used when handling the two exceptions mentioned above, since there is no use then for adding the Abox statement again.